src/main/java/com/mraof/minestuck/world/gen/LandChunkGenerator.java

```diff
@@ -128,4 +132,33 @@ private static boolean hasGatePlacement(ChunkGeneratorStructureState state, Hold
 	{
 		return state.getPlacementsForStructure(structure).stream().anyMatch(placement -> placement.type() == MSStructurePlacements.LAND_GATE.get());
 	}
+	
+	public void init(LandTypeExtensions extensions)
+	{
+		this.customBiomeSettings.init(extensions);
+	}
+	
+	private static final class LazyBiomeSettings
+	{
+		private final Function<LandTypeExtensions, LandCustomBiomeSettings> constructor;
+		private LandCustomBiomeSettings value = null;
+		
+		
+		public LazyBiomeSettings(Function<LandTypeExtensions, LandCustomBiomeSettings> constructor)
+		{
+			this.constructor = constructor;
+		}
+		
+		public LandCustomBiomeSettings get()
+		{
+			return Objects.requireNonNull(this.value);
```

Comment:
Crash can occur here when a Land which previously had no custom mobs is suddenly given some. 
Tracing its usage back to line 107.

`else return customBiomeSettings.get().customMobSpawnsFor(biome).getMobs(category);`

Reply:
I don't think I am following you. I'm not sure lands are ever "suddenly given custom mobs", with land mob spawns only being set up/loaded at either server start or land creation, with no regard to what it might've been before. I also don't think there being 0 or 1 custom mobs will make a difference anywhere other than in the return of `getMobs()`.
Did you get a crash at that point? If so, the issue may actually be something else.